beetsplug/unimported.py

```diff
@@ -53,7 +53,7 @@ def print_unimported(lib, opts, args):
                 for file in f
                 if not any(
                     [file.endswith(ext) for ext in ignore_exts]
-                    + [r in ignore_dirs]
+                    + [r.startswith(ignore_dir) for ignore_dir in ignore_dirs]
```

Comment:
It's probably a rare scenario, but it'd be good to handle the case where an ignored directory matches the prefix of a directory that the user doesn't want to ignore (e.g. with this change, ignoring `abc` would also ignore the directory `abcdef`).

We'd probably need some special handling to check if the name starts with `ignore_dir` followed by a path separator?